tests/unit/fem/test_get_value.cpp

```diff
@@ -1829,4 +1904,1669 @@ TEST_CASE("3D GetVectorValue",
+               REQUIRE(rt_err == Approx(0.0));
+               REQUIRE(dgv_err == Approx(0.0));
+            }
+         }
+
+         SECTION("Boundary Evaluation 2D (DG Context)")
+         {
+            std::cout << "Boundary Evaluation 2D (DG Context)" << std::endl;
+            for (int be = 0; be < mesh.GetNBE(); be++)
+            {
+               FaceElementTransformations *T =
+                  mesh.GetBdrFaceTransformations(be);
+               const IntegrationRule &ir = IntRules.Get(T->GetGeometryType(),
+                                                        2*order + 2);
+
+               double h1_err = 0.0;
+               double rt_err = 0.0;
+               double dgv_err = 0.0;
+
+               double tip_data[dim];
+               Vector tip(tip_data, dim);
+               for (int j=0; j<ir.GetNPoints(); j++)
+               {
+                  npts++;
+                  const IntegrationPoint &ip = ir.IntPoint(j);
+
+                  T->SetIntPoint(&ip);
+                  T->Transform(ip, tip);
+
+                  double f_val = DivFunc_2D_quad(tip);
+
+                  double h1_gf_val = h1_xCoef.Eval(*T, ip);
+                  double rt_gf_val = rt_xCoef.Eval(*T, ip);
+                  double dgv_gf_val = dgv_xCoef.Eval(*T, ip);
+
+                  h1_err += fabs(f_val - h1_gf_val);
+                  rt_err += fabs(f_val - rt_gf_val);
+                  dgv_err += fabs(f_val - dgv_gf_val);
+
+                  if (log > 0 && fabs(f_val - h1_gf_val) > tol)
+                  {
+                     std::cout << be << ":" << j << " h1  " << f_val << " "
+                               << h1_gf_val << " " << fabs(f_val - h1_gf_val)
+                               << std::endl;
+                  }
+                  if (log > 0 && fabs(f_val - rt_gf_val) > tol)
+                  {
+                     std::cout << be << ":" << j << " rt  " << f_val << " "
+                               << rt_gf_val << " " << fabs(f_val - rt_gf_val)
+                               << std::endl;
+                  }
+                  if (log > 0 && fabs(f_val - dgv_gf_val) > tol)
+                  {
+                     std::cout << be << ":" << j << " dgv " << f_val << " "
+                               << dgv_gf_val << " " << fabs(f_val - dgv_gf_val)
+                               << std::endl;
+                  }
+               }
+               h1_err /= ir.GetNPoints();
+               rt_err /= ir.GetNPoints();
+               dgv_err /= ir.GetNPoints();
+
+               REQUIRE(h1_err == Approx(0.0));
+               REQUIRE(rt_err == Approx(0.0));
+               REQUIRE(dgv_err == Approx(0.0));
+            }
+         }
+      }
+   }
+   std::cout << "Checked GridFunction::GetDivergence at "
+             << npts << " 2D points" << std::endl;
+}
+
+TEST_CASE("3D GetDivergence",
+          "[GridFunction]"
+          "[DivergenceGridFunctionCoefficient]")
+{
+   int log = 1;
+   int n = 1;
```

Comment:
I noticed most tests use a 1-element mesh. Does it make sense to test on at least 2 elements, and maybe one of them locally refined?

Reply:
Honestly, I was a little hesitant to use such a small mesh.  However, my justification is two-fold: I want to keep the unit tests as quick as possible and the only thing that is likely to break on a larger mesh would be the element transformation which is tested elsewhere.

I can certainly increase the mesh sizes if you like.  After all, the tests are meant to find errors that we don't predict.